Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -450,6 +467,63 @@ private static void WriteAutoSwapOngoing()
             }
         }
 
+        /// <summary>
+        /// This works around the hostname dns resolution issue for recently created site.
+        /// If dns failed, we will use the home hosted service as alternative IP address.
+        /// </summary>
+        /// <returns></returns>
+        private static async Task<IPAddress> GetAlternativeIPAddress(string host)
+        {
+            try
+            {
+                // if resolved successfully, return null to not use alternative ipAddress
+                await Dns.GetHostEntryAsync(host);
+                return null;
+            }
+            catch (Exception ex)
+            {
+                Trace(TraceEventType.Verbose, "Unable to dns resolve {0}.  {1}", host, ex);
+            }
+
+            return await GetHomeStampAddress(host);
+        }
+
+        private static async Task<IPAddress> GetHomeStampAddress(string host)
+        {
+            var homeStamp = HomeStamp;
+            if (string.IsNullOrEmpty(homeStamp))
+            {
+                return null;
+            }
+
+            var homeStampHostName = string.Format("{0}.cloudapp.net", homeStamp);
+            if (host.EndsWith(".scm.azurewebsites.us", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.usgovcloudapp.net", homeStamp);
+            }
+            else if (host.EndsWith(".scm.chinacloudsites.cn", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.chinacloudapp.cn", homeStamp);
+            }
+            else if (host.EndsWith(".scm.azurewebsites.de", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.azurecloudapp.de", homeStamp);
+            }
+
+            try
+            {
+                Trace(TraceEventType.Verbose, "Try to dns resolve {0}.", homeStampHostName);
+                var entry = await Dns.GetHostEntryAsync(homeStampHostName);
+                return entry.AddressList.First();
+            }
+            catch (Exception ex)
+            {
+                Trace(TraceEventType.Verbose, "Unable to dns resolve {0}.  {1}", homeStampHostName, ex);
```

Comment:
Make this trace different from the one above, so we can more easily differentiate when searching logs. e.g. `Unable to dns resolve stamp {0}`

Reply:
Added.    BTW, the first arg "{0}" will represent stamp hostname - so it will quite obvious one way or another.   Below are the sample output traces.

```
PostDeployment Verbose: 16 : Syncing httpTrigger of function1
PostDeployment Verbose: 16 : Syncing orchestrationTrigger of function1
PostDeployment Verbose: 16 : Syncing activityTrigger of function1
PostDeployment Verbose: 16 : Syncing queueTrigger of function2
PostDeployment Verbose: 16 : Function function3 is disabled
PostDeployment Verbose: 16 : Function function4 is excluded
PostDeployment Verbose: 16 : Function function5 is disabled
PostDeployment Verbose: 16 : Unable to dns resolve suwatchet01nogood.scm.kudu1.antares-test.windows-int.net.  System.Net.Sockets.SocketException (0x80004005): No such host is known
   at System.Net.Dns.HostResolutionEndHelper(IAsyncResult asyncResult)
   at System.Net.Dns.EndGetHostEntry(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Kudu.Core.Helpers.PostDeploymentHelper.<GetAlternativeIPAddress>d__36.MoveNext() in C:\gitroot\kudu\Kudu.Core\Helpers\PostDeploymentHelper.cs:line 480
PostDeployment Verbose: 16 : Try to dns resolve kudu1.cloudapp.net.
PostDeployment Verbose: 16 : Begin HttpPost https://23.96.210.137/operations/settriggers, host: suwatchet01nogood.scm.kudu1.antares-test.windows-int.net, x-ms-request-id: c553cad9-38af-4273-8047-0302d281634b
PostDeployment Verbose: 16 : End HttpPost, status: OK
PostDeployment Information: 8 : Syncing 5 function triggers with payload size 434 bytes successful.
```